TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/TASingleton/TASingleton.m

```diff
@@ -0,0 +1,596 @@
+//
+//  TASingleton.m
+//  TestAssignmentDoItSoftware
+//
+//  Created by Vladimir Samoylenko on 2/15/19.
+//  Copyright © 2019 TheRedQueen. All rights reserved.
+//
+
+#import "TASingleton.h"
+
+@implementation TASingleton
+
+@synthesize managedObjectModel;
+@synthesize managedObjectContext;
+@synthesize persistentContainer = _persistentContainer;
+@synthesize userInfo;
+
+#pragma mark - singleton global procedures
+
++ (TASingleton *)sharedInstance{
+
+    static TASingleton *sharedInstance = nil;
+    
+    static dispatch_once_t predicate;
+    dispatch_once( &predicate, ^{
+        sharedInstance = [[ self alloc ] init];
+    });
+    
+    return sharedInstance;
+}
+
+#pragma mark - Core Data procedures
+
+-(NSNumber *)getNewId{
+    
+    NSDate *from1970 = [[NSDate alloc] initWithTimeIntervalSinceNow:0];
+    NSNumber *newid = [NSNumber numberWithUnsignedInteger:[[NSNumber numberWithUnsignedInteger:[from1970 timeIntervalSince1970]] unsignedIntegerValue]];
+
+    return newid;
```

Comment:
Timestamps make pretty bad IDs in general. It sounds like it should fit the task at hand but it's not obvious what the implications are. Is this going to fail when e.g. a new feature is added to create tasks from a text file?